org.eclipse.dartboard/src/org/eclipse/dartboard/project/DartProjectPage.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2019 vogella GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Lakshminarayana Nekkanti 
+ *******************************************************************************/
+package org.eclipse.dartboard.project;
+
+import org.eclipse.dartboard.Messages;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.dialogs.WizardNewProjectCreationPage;
+
+public class DartProjectPage extends WizardNewProjectCreationPage {
+
+	public DartProjectPage(String pageName) {
+		super(pageName);
+	}
+
+	@Override
+	public void createControl(Composite parent) {
+		super.createControl(parent);
+		createAdditionalControls((Composite) getControl());
+	}
+
+	private void createAdditionalControls(Composite parent) {
+		Group dartGroup = new Group(parent, SWT.NONE);
+		dartGroup.setFont(parent.getFont());
+		dartGroup.setText(Messages.NewProject_Group_Label);
+		dartGroup.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+		dartGroup.setLayout(new GridLayout(2, false));
+
+		Label label = new Label(dartGroup, SWT.NONE);
+		label.setText(Messages.NewProject_SDK_Version_Label);
+
+		Combo combo = new Combo(dartGroup, SWT.READ_ONLY | SWT.BORDER);
+		GridData textData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+		textData.horizontalIndent = 0;
+		combo.setLayoutData(textData);

```

Comment:
I just realized, that you're not filling in any data into the combo. Since for now the preference page only allows one dart sdk location, this combo shouldn't be shown. Please remove the combo box and maybe print the string to the path of the SDK here as a label (e.g. a non changeable text). See LaunchConfigTab#91 on how to get the current location of the Sdk.

Reply:
![image](https://user-images.githubusercontent.com/39690425/60399257-7eae1300-9b7f-11e9-9119-a5098149acc9.png)

Is it okay for you? If the SDK location is not set, what should we show?. Just blank ? or any message?

Reply:
Yes, that looks good.

> If the SDK location is not set, what should we show?. Just blank ? or any message?

Maybe we could let the user set the location from this page as well, using a button if the SDK is not set on the preference page.
But for me it would be enough to just show a message that the SDK location is not set and that the user should visit the preference page for now.

Reply:
Can you write rephrase the error message here?
"Could not find the default Dart SDK location. Please set it in a preference page."

Reply:
"Dart SDK not found. Please select a valid SDK location on the preference page."  should do.

As I said this is more of temporary solution.